elyra/pipeline/kfp/processor_kfp.py

```diff
@@ -510,15 +510,15 @@ def _cc_pipeline(self,
         runtime_configuration = self._get_metadata_configuration(schemaspace=Runtimes.RUNTIMES_SCHEMASPACE_ID,
                                                                  name=pipeline.runtime_config)
 
-        cos_endpoint = runtime_configuration.metadata['cos_endpoint']
-        cos_username = runtime_configuration.metadata['cos_username']
-        cos_password = runtime_configuration.metadata['cos_password']
+        cos_endpoint = runtime_configuration.metadata.get('cos_endpoint')
```

Comment:
Did you mean to treat this as an optional property? The KFP and Airflow schema updates you've made still list `cos_endpoint` as required and the existing code `cos_endpoint = runtime_configuration.metadata['cos_endpoint']` would therefore be more appropriate.

Reply:
I was making this consistent with the [`processor_airflow.py`](https://github.com/elyra-ai/elyra/blob/243dab7a22761ef71c1f34cce0bf4828cdd8e130/elyra/pipeline/airflow/processor_airflow.py#L94).

NOTE: this does not change the fact that `cos_endpoint` is still required, as it is specified under [`metadata/schemas/kfp.json`](https://github.com/elyra-ai/elyra/blob/25ffbe35965426b89bd91bbf69660959bcec6846/elyra/metadata/schemas/kfp.json#L162)

Reply:
Got it! I am all for consistency. However, I confirmed with Alan that the referenced Airflow code should have used `['..']` instead of `.get('')` to access required properties. Since the metadata service enforces the required constraint the existing code should never fail. Can you please revert the change to avoid the impression that this particular property is optional?

Reply:
Done in https://github.com/elyra-ai/elyra/pull/2335/commits/4f057142900ccfd0c4502ac493b5549ef230dd7d